channels/urbdrc/client/libusb/libusb_udevman.c

```diff
@@ -542,18 +560,21 @@ static int hotplug_callback(struct libusb_context* ctx, struct libusb_device* de
 	if (rc != LIBUSB_SUCCESS)
 		return rc;
 
-	if (device_is_filtered(dev, &desc, event))
-		return 0;
-
 	switch (event)
 	{
 		case LIBUSB_HOTPLUG_EVENT_DEVICE_ARRIVED:
-			if (idevman->isAutoAdd(idevman))
-				add_device(idevman, DEVICE_ADD_FLAG_ALL, bus, addr, desc.idVendor, desc.idProduct);
+			pair.vid = desc.idVendor;
+			pair.pid = desc.idProduct;
+			if (ArrayList_Contains(udevman->hotplug_vid_pids, &pair) ||
+			    (udevman->iface.isAutoAdd(&udevman->iface) &&
+			     !device_is_filtered(dev, &desc, event)))
+				add_device(&udevman->iface, DEVICE_ADD_FLAG_ALL, bus, addr, desc.idVendor,
```

Comment:
Braces could improve readability here

Reply:
But CloseHandle already checks for null pointers itself